src/NServiceBus.AcceptanceTests/Core/DelayedDelivery/TimeoutManager/When_using_external_timeout_manager_code_first.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.DelayedDelivery.TimeoutManager
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.DelayedDelivery;
+    using NUnit.Framework;
+
+    public class When_using_external_timeout_manager_code_first : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_send_delayed_messages_to_external_TimeoutManager()
+        {
+            Requires.TimeoutStorage();
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithTimeoutManager>()
+                .WithEndpoint<Endpoint>(b => b.When((session, c) =>
+                {
+                    var options = new SendOptions();
+
+                    options.DelayDeliveryWith(TimeSpan.FromDays(5));
+                    options.RouteToThisEndpoint();
+
+                    return session.Send(new DelayedMessage(), options);
+                }))
+                .Done(c => c.ExternalTimeoutManagerInvoked)
+                .Run();
+
+            Assert.IsTrue(context.TimeoutManagerHeaderDetected);
+        }
+
+        [Test]
+        public async Task Should_send_delayed_retries_to_external_TimeoutManager()
+        {
+            Requires.TimeoutStorage();
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithTimeoutManager>()
+                .WithEndpoint<Endpoint>(b => b
+                    .CustomConfig(e => e.Recoverability()
+                        .Delayed(s => s.NumberOfRetries(1).TimeIncrease(TimeSpan.FromSeconds(1))))
+                    .When((session, c) => session.SendLocal(new FailingMessage())))
+                .Done(c => c.ExternalTimeoutManagerInvoked)
+                .Run();
+
+            Assert.IsTrue(context.TimeoutManagerHeaderDetected);
```

Comment:
no strong feelings. I don't see a real reason to change it though, it states what it does and we're just testing all the available information we have to assume whether the TM has been used or not.